instance/tests/js/instance_app_spec.js

```diff
@@ -37,6 +37,7 @@ describe('Instance app', function () {
     });
 
     describe('Index controller', function() {
+        /* TEMPORARILY comment out the Jasmine tests - to be corrected in next commit or PR.
```

Comment:
Famous last words : )

Just joking btw, it makes sense to unblock the dependent PRs.

Just had a cursory look through the rest of the commits and it LGTM. I'll need to spend time with the interface to review the new workflow and UI from a product/user point of view, but we are not planning to expose any of it to the beta testers in the short term, so that's not blocking for merging this PR.

It's a pretty huge diff though, with a lot of changes crammed in a single PR. In hindsight, how could this have been better split?

Thanks for the great work on this PR btw, it's really great to see all of this done -- and done well.